projects/opendr_ws_2/src/opendr_perception/opendr_perception/panoptic_segmentation_efficient_ps.py

```diff
@@ -0,0 +1,192 @@
+
+            # The output topics are only published if there is at least one subscriber
+            if self._visualization_publisher is not None and self._visualization_publisher.get_subscription_count() > 0:
+                panoptic_image = EfficientPsLearner.visualize(image, prediction, show_figure=False,
+                                                              detailed=self.detailed_visualization)
+                self._visualization_publisher.publish(self._bridge.to_ros_image(panoptic_image, encoder="rgb8"))
+
+            if self._instance_heatmap_publisher is not None and self._instance_heatmap_publisher.get_subscription_count() > 0:
+                self._instance_heatmap_publisher.publish(self._bridge.to_ros_image(prediction[0]))
+            if self._semantic_heatmap_publisher is not None and self._semantic_heatmap_publisher.get_subscription_count() > 0:
+                self._semantic_heatmap_publisher.publish(self._bridge.to_ros_image(prediction[1]))
+
+        except Exception as e:
+            self.get_logger().error(f'Failed to generate prediction: {e}')
+
+
+def main(args=None):
+    rclpy.init(args=args)
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-i', '--input_rgb_image_topic', type=str, default='/image_raw',
+                        help='listen to RGB images on this topic')
+    parser.add_argument('--checkpoint', type=str, default='cityscapes',
+                        help='download pretrained models [cityscapes, kitti] or load from the provided path')
+    parser.add_argument('-o', '--output_heatmap_topic', type=str, default='/opendr/panoptic',
+                        help='publish the semantic and instance maps on this topic as "OUTPUT_HEATMAP_TOPIC/semantic" \
+                                 and "OUTPUT_HEATMAP_TOPIC/instance"')
+    parser.add_argument('-v', '--output_rgb_image_topic', type=str,
+                        default='/opendr/panoptic/rgb_visualization',
+                        help='publish the panoptic segmentation map as an RGB image on this topic or a more detailed \
+                                  overview if using the --detailed_visualization flag')
+    parser.add_argument('--detailed_visualization', action='store_true',
+                        help='generate a combined overview of the input RGB image and the semantic, instance, and \
+                                  panoptic segmentation maps and publish it on OUTPUT_RGB_IMAGE_TOPIC')
+    args = parser.parse_args()
+    efficient_ps_node = EfficientPsNode(args.input_rgb_image_topic,
+                                        args.checkpoint,
+                                        args.output_heatmap_topic,
+                                        args.output_rgb_image_topic,
+                                        args.detailed_visualization)
+    efficient_ps_node.listen()
+if __name__ == '__main__':
+    main()
```

Comment:
Some PEP8 missing newlines

```suggestion
    efficient_ps_node.listen()


if __name__ == '__main__':
    main()
```